client/Windows/cli/wfreerdp.c

```diff
@@ -66,9 +67,24 @@ INT WINAPI WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, LPSTR lpCmdLine
 
 	context->argc = __argc;
 	context->argv = (char**) malloc(sizeof(char*) * __argc);
+	if (!context->argv)
+	{
+		ret = 1;
+		goto out;
+	}
 
 	for (index = 0; index < context->argc; index++)
+	{
 		context->argv[index] = _strdup(__argv[index]);
+		if (!context->argv[index])
+		{
+			ret = 1;
+			free(context->argv);
```

Comment:
Since you're freeing argv I wonder why the array members (up to index-1) members are not freed here.
However, since this is WinMain the usefulness is questionable anyways.